patches/api/0460-add-rich-message-component-support-to-configuration.patch

```diff
@@ -0,0 +1,85 @@
+Subject: [PATCH] add rich message component support to configuration
+
+
+diff --git a/src/main/java/org/bukkit/configuration/ConfigurationSection.java b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+index b6b00af08f12f838411845e4f4e29e62826dfc7f..9dc4065df623cb849b3e96dc4a6c7ce757525c82 100644
+--- a/src/main/java/org/bukkit/configuration/ConfigurationSection.java
++++ b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+@@ -1058,4 +1058,54 @@ public interface ConfigurationSection {
+      * one line.
+      */
+     public void setInlineComments(@NotNull String path, @Nullable List<String> comments);
++
++    // Paper start - add rich message component support to configuration
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @return Requested Component.
++     */
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path) {
++        return getRichMessage(path, null);
++    }
++
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param component component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, !null -> !null")
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path, @Nullable net.kyori.adventure.text.Component component){
++        return getComponent(path, net.kyori.adventure.text.minimessage.MiniMessage.miniMessage(), component);
++    }
++
++    /**
++     * Gets the requested Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param componentSerializer componentSerializer instance used for deserialization
++     * @param component component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, _, !null -> !null")
++    @Nullable
++    net.kyori.adventure.text.Component getComponent(@NotNull String path, @NotNull net.kyori.adventure.text.serializer.ComponentSerializer<net.kyori.adventure.text.Component, net.kyori.adventure.text.Component, String> componentSerializer, @Nullable net.kyori.adventure.text.Component component);
```

Comment:
This method should use generics to support serializers that output other Component subtypes